### ISSUE TITLE
[SDK] GetApplications Method

### ISSUE THREAD

**Issue by ezeike** [pokt-network/poktroll#324]
## Objective

Introduce a `GetApplications` method to the `Shannon SDK`.

## Origin Document

The Grove gateway requires a list of all staked applications in order to fetch all sessions and perform QoS on all in-session nodes.

The screenshot below captures the start of this discussion:

<img width="940" alt="Screenshot 2024-01-16 at 11 59 23 AM" src="https://github.com/pokt-network/poktroll/assets/1892194/9e29eaee-6d61-4ae5-a08e-18fb1aac8829">

## Goals

- Allow SDK users to get a list of staked applications
- Enable continued integration of the GROVE QoS module with the Shannon SDK

## Deliverables

- [ ] 1. An initial investigation into [fern-api/fern](https://github.com/fern-api/fern) to see if it could easily provide the SDK we need for integration
2. After the above is done:
	- [ ] A single PR in this repository that introduces the `GetApplications` SDK Method
3. After the above is done:
	- [ ] A single PR in the Portal repository that uses the newly introduced method

## Non-goals / Non-deliverables

- Functioning Grove gateway QoS

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @ezeike 
**Co-owner**: @Olshansk @red-0ne

**Comment by Olshansk** [comment]
@ezeike Made some edits to the ticket, lmk if 👍 ?

**Comment by ezeike** [comment]
@Olshansk completed the fern experiment.

It turned out that the REST API exposed by the Cosmos SDK is not OpenAPI [compliant.](https://github.com/pokt-network/poktroll/pull/344#issuecomment-1913350721)

The rest of this ticket has been superseded by #339

**Comment by Olshansk** [comment]
> It turned out that the REST API exposed by the Cosmos SDK is not OpenAPI https://github.com/pokt-network/poktroll/pull/344#issuecomment-1913350721

👍 

> The rest of this ticket has been superseded by https://github.com/pokt-network/poktroll/pull/339

@ezeike It has not been superseded by #339, but rather is being resolved by it i.

This is a GitHub issue (a task), and 339 is a GitHub PR (a solution). Once the PR is merged in, this ticket can be resolved.

**Comment by ezeike** [comment]
#339 has been merged, closing ticket.
